BitwardenShared/UI/Auth/Utilities/VaultUnlockSetupHelper.swift

```diff
@@ -0,0 +1,153 @@
+// MARK: - VaultUnlockSetupHelper
+
+/// A protocol for a helper used to set up vault unlock methods.
+///
+protocol VaultUnlockSetupHelper: AnyObject {
+    /// Enables or disables biometric vault unlock.
+    ///
+    /// - Parameters:
+    ///   - enabled: Whether to enable biometric vault unlock.
+    ///   - showAlert: A closure used to handle showing an alert if needed while toggling biometric
+    ///     vault unlock.
+    /// - Returns: The biometric unlock status after vault unlock has been updated.
+    ///
+    func setBiometricUnlock(enabled: Bool, showAlert: @escaping (Alert) -> Void) async -> BiometricsUnlockStatus?
+
+    /// Enables or disables pin vault unlock.
+    ///
+    /// - Parameters:
+    ///   - enabled: Whether to enable pin vault unlock.
+    ///   - showAlert: A closure used to handle showing an alert if needed while toggling pin vault unlock.
+    /// - Returns: Whether pin unlock is enabled after vault unlock has been updated.
+    ///
+    func setPinUnlock(enabled: Bool, showAlert: @escaping (Alert) -> Void) async -> Bool
+}
+
+// MARK: - VaultUnlockSetupHelperError
+
+/// Errors thrown by a `VaultUnlockSetupHelper`.
+///
+enum VaultUnlockSetupHelperError: Error {
+    /// The user cancelled setting up an unlock method.
+    case userCancelled
+}
+
+// MARK: - DefaultVaultUnlockSetupHelper
+
+/// A default implementation of `VaultUnlockSetupHelper` which is used to set up vault unlock methods.
+///
+@MainActor
+class DefaultVaultUnlockSetupHelper {
```

Comment:
just curious, why not make this a service itself?

Reply:
We've used this concept of a helper elsewhere to extract logic from the processor or share logic between multiple processors. In this case, while not directly showing alerts, it does end up creating alerts that should be shown. So I feel like it belongs in the UI layer rather than the service layer.